File: src/IServices/IGarageService.ts
import { ICoordenada } from "../Infraestructure/Interfaces/ICoordenada";
import { IGarage } from "../Infraestructure/Interfaces/IGarage";
import { IVehiculo } from "../Infraestructure/Interfaces/IVehiculo";

export interface IGarageService {

    EntrarVehiculo(vehiculo: IVehiculo, piso: number, posicion: number): ICoordenada | null; 
    EntrarVehiculo(vehiculo: IVehiculo): ICoordenada | null; 
    
    QuitarVehiculo(piso: number, posicion: number): number;
    GetPrimerLugarLibre(): ICoordenada | null;
}

File: src/main/java/Grupo2/Garage/IServices/IGarageService.java
package Grupo2.Garage.IServices;

import Grupo2.Garage.Infraestructure.Class.Garage.Coordenada;
import Grupo2.Garage.Infraestructure.Exceptions.GarageException;
import Grupo2.Garage.Infraestructure.Interfaces.IVehiculo;

public interface IGarageService {
    public Coordenada GuardarVehiculoPosicion(IVehiculo vehiculo, Integer piso, Integer posicion) throws GarageException;
    public Coordenada GuardarVehiculo(IVehiculo vehiculo ) throws GarageException;

    public Double QuitarVehiculo(Integer piso, Integer posicion, String patente) throws GarageException;
    public Coordenada GetPrimerLugarLibre() throws GarageException;
}
